src/components/application_manager/src/state_controller.cc

```diff
@@ -66,29 +66,56 @@ void StateController::HmiLevelConflictResolver::operator ()
   using namespace mobile_apis;
   using namespace helpers;
   DCHECK_OR_RETURN_VOID(state_ctrl_);
-  if (to_resolve == applied_)
-    return;
-  if (Compare<HMILevel::eType, EQ, ONE>(state_->hmi_level(),
-                                        HMILevel::HMI_FULL,
-                                        HMILevel::HMI_LIMITED)) {
-    HmiStatePtr cur_state = to_resolve->RegularHmiState();
-    if (Compare<HMILevel::eType, EQ, ONE>(cur_state->hmi_level(),
-                                          HMILevel::HMI_FULL,
-                                          HMILevel::HMI_LIMITED)) {
-      if (applied_->IsAudioApplication() && state_ctrl_->IsSameAppType(applied_, to_resolve)) {
-        state_ctrl_->SetupRegularHmiState(to_resolve, HMILevel::HMI_BACKGROUND,
-                     AudioStreamingState::NOT_AUDIBLE);
-      } else if (HMILevel::HMI_FULL == state_->hmi_level() &&
-          HMILevel::HMI_FULL == cur_state->hmi_level()) {
-        LOG4CXX_DEBUG(logger_, "HMI level will be changed for application with appID="
-                      <<to_resolve->app_id());
-        state_ctrl_->SetupRegularHmiState(to_resolve, HMILevel::HMI_LIMITED,
-                     AudioStreamingState::AUDIBLE);
-      } else {
-        LOG4CXX_DEBUG(logger_, "HMI will not be changed level for application with appID="
-                      <<to_resolve->app_id());
-      }
-    }
+  if (to_resolve == applied_) return;
```

Comment:
This code is copy/paste from CustomSDL/sdl_panasonic/develop(451042b). May be it is better to keep it the same for future comparison.
Also this return is about skipping the current app and updating all others. It is called in a cycle over all apps. This is not really a special case - the purpose is to update the other apps.